Modules/Devices/src/Devices.Infrastructure.Database.Postgres/Migrations/20230818153325_AppIdForPnsRegistrations.cs

```diff
@@ -2,47 +2,48 @@
 
 #nullable disable
 
-namespace Backbone.Modules.Devices.Infrastructure.Database.Postgres.Migrations
+namespace Backbone.Modules.Devices.Infrastructure.Database.Postgres.Migrations;
+
+/// <inheritdoc />
+public partial class AppIdForPnsRegistrations : Migration
 {
     /// <inheritdoc />
-    public partial class AppIdForPnsRegistrations : Migration
+    protected override void Up(MigrationBuilder migrationBuilder)
+    {
+        migrationBuilder.DropForeignKey(
+            name: "FK_Identities_Tiers_TierId",
+            table: "Identities");
+
+        migrationBuilder.DropIndex(
+            name: "IX_Identities_TierId",
+            table: "Identities");
+
+        migrationBuilder.AddColumn<string>(
+            name: "AppId",
+            table: "PnsRegistrations",
+            type: "text",
+            nullable: false);
+
+        migrationBuilder.Sql(""" DELETE FROM "PnsRegistrations" WHERE "AppId" IS NULL """);
```

Comment:
Mhh, I told you to FIRST delete all registrations, and THEN add the AppId column. Currently it will fail because you add a non-nullable column even though there are rows that don't have a value for that column and there is no default value.